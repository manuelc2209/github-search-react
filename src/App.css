.App {
  text-align: center;
}

#root {
  background-color: #232324;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: rgb(255, 255, 255);
}

body {
  margin: 0;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

html,
body,
* {
  font-family: monospace;
}
