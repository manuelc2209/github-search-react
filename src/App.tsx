import React from 'react';
import { UIComponent } from './components/ui/InitialPage';
import { StyledContainer, StyledHeader } from './styles';

function App() {
    return (
        <StyledContainer>
            <StyledHeader>
                <UIComponent />
            </StyledHeader>
        </StyledContainer>
    );
}

export default App;
