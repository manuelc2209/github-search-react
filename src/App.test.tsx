import { render } from '@testing-library/react';
import App from './App';

vi.mock('react-router-dom', () => {
    const originalModule = vi.importActual('react-router-dom');

    return {
        __esModule: true,
        ...originalModule,
        useNavigate: vi.fn(),
        useLocation: vi.fn().mockReturnValue({ pathname: '/' })
    };
});

test('Should render the app', () => {
    const view = render(<App />);
    expect(view).toMatchSnapshot();
});
