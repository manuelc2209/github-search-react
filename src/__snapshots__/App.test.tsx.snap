// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="sc-iwjdpV lhbYkA"
        >
          <div>
            <div>
              <h1>
                Search for GitHub Users
              </h1>
              <div
                class="sc-furwcr jrFWNp"
              >
                <input
                  class="sc-pVTFL lktBfo"
                  data-testid="StyledInput"
                  placeholder="Search a user or organization"
                  value=""
                />
              </div>
            </div>
            <div>
              Enter a login, name or company you are looking for
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="sc-iwjdpV lhbYkA"
      >
        <div>
          <div>
            <h1>
              Search for GitHub Users
            </h1>
            <div
              class="sc-furwcr jrFWNp"
            >
              <input
                class="sc-pVTFL lktBfo"
                data-testid="StyledInput"
                placeholder="Search a user or organization"
                value=""
              />
            </div>
          </div>
          <div>
            Enter a login, name or company you are looking for
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
